src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -6681,12 +6681,17 @@ public override DynamicMetaObject FallbackInvokeMember(DynamicMetaObject target,
                         .WriteToDebugLog(this);
                     BindingRestrictions argRestrictions = args.Aggregate(BindingRestrictions.Empty, static (current, arg) => current.Merge(arg.PSGetMethodArgumentRestriction()));
 
-                    if (string.Equals(Name, "Where", StringComparison.OrdinalIgnoreCase))
+                    if (string.Equals(Name, "Where", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(Name, "PSWhere", StringComparison.OrdinalIgnoreCase))
                     {
                         return InvokeWhereOnCollection(emptyEnumerator, args, argRestrictions).WriteToDebugLog(this);
                     }
 
-                    if (string.Equals(Name, "ForEach", StringComparison.OrdinalIgnoreCase))
+                    if (string.Equals(Name, "ForEach", StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(Name, "PSForEach", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // We need to pass the empty enumerator to the ForEach operator, so that it can return an empty collection.
+                        // The ForEach operator will not be able to call the script block if the enumerator is empty.
                     {
```

Comment:
```suggestion
                    if (string.Equals(Name, "ForEach", StringComparison.OrdinalIgnoreCase)
                        || string.Equals(Name, "PSForEach", StringComparison.OrdinalIgnoreCase))
                    {
                        // We need to pass the empty enumerator to the ForEach operator, so that it can return an empty collection.
                        // The ForEach operator will not be able to call the script block if the enumerator is empty.
```